lib/ansible/modules/cloud/misc/virt.py

```diff
@@ -583,6 +580,9 @@ def main():
             uri=dict(type='str', default='qemu:///system'),
             xml=dict(type='str'),
         ),
+        required_by=dict(
+            state=('name'),
```

Comment:
`name` seems not to be required in case `command == 'list_vms'` and `state` is given. (See lines 470 to 474.)

Reply:
This was wrong in the original implementation too AFAICT.